src/v/cloud_storage/offset_translation_layer.h

```diff
@@ -26,15 +28,19 @@ namespace cloud_storage {
 /// It consumes information stored in the manifest.
 class offset_translator final {
 public:
-    explicit offset_translator(model::offset initial_delta)
-      : _initial_delta(initial_delta) {}
+    explicit offset_translator(
+      model::offset initial_delta, ss::abort_source& as)
```

Comment:
In all other changes in this PR we have optional reference, but due to the type this reference is not optional. 

Should the type here be `storage::opt_abort_source_t` for consistency?

Reply:
done